### ISSUE TITLE
RawApplePayButton's onPressed doesn't work with VoiceOver on iOS 16.5 

### ISSUE THREAD

**Issue by romgrm** [google-pay/flutter-plugin#208]
Hi ! 

In the video showing the bug, I have 2 RawApplePayButtons which are supposed to open a bottomSheet when onPressed is triggered. 

https://github.com/google-pay/flutter-plugin/assets/72022928/d4e4c3b3-5eeb-49f2-a911-97e60ec2701e

The first Apple Pay button is used "conventionally", i.e. there are no other Semantics wrapped around it, so the widget works fine.

The second button is wrapped by a SizedBox(), itself wrapped by a Semantics(). When the RawApplePayButton is tapped, the callback is not triggered.
However, as you can see in the video, if I focus on my SizedBox() and click on it, the onPressed on my RawApplePayButton is called, which shouldn't be the case. 

In my Apple Pay button KO, the focus seems to be directly on the PKPaymentButton and so Flutter's onPressed callback isn't triggered. 

The bug seems to be only from iOS 16.5 onwards, and only when voice is used. (I tested on iPhone 14 pro on 16.5.1)

Here is the code to reproduce if you wish : 
```dart
class ApplePayIssue extends StatefulWidget {
  const ApplePayIssue({super.key});

  @override
  State<ApplePayIssue> createState() => _ApplePayIssueState();
}

class _ApplePayIssueState extends State<ApplePayIssue> {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
        body: Center(
      child: Column(
        mainAxisAlignment: MainAxisAlignment.center,
        children: [
          const Text("apple pay button OK"),
          RawApplePayButton(
            style: ApplePayButtonStyle.black,
            type: ApplePayButtonType.plain,
            onPressed: () {
              showModalBottomSheet(
                  context: context,
                  builder: (context) {
                    return const SizedBox(
                      height: 200,
                      child: Text("apple pay modal OK"),
                    );
                  });
            },
          ),
          const SizedBox(
            height: 50,
          ),
          const Text("apple pay button KO"),
          Semantics(
            label: "paiement card",
            excludeSemantics: false,
            explicitChildNodes: true,
            child: Container(
              color: Colors.red,
              padding: const EdgeInsets.all(10),
              child: RawApplePayButton(
                style: ApplePayButtonStyle.black,
                type: ApplePayButtonType.plain,
                onPressed: () {
                  showModalBottomSheet(
                      context: context,
                      builder: (context) {
                        return const SizedBox(
                          height: 200,
                          child: Text("apple pay modal OK"),
                        );
                      });
                },
              ),
            ),
          ),
        ],
      ),
    ));
  }
}
```

Do you think this is due to the plugin or the framework?  

Thanks for the help :)

**Comment by JlUgia** [comment]
Hey @romgrm, the way the library draws the Apple Pay button is fairly standard, so I'd start looking within the platform first.
Take a look at [this issue](https://github.com/flutter/flutter/issues/113377), which seems related.
I'm closing the issue for now. Feel free to reopen if you think these two issues are disconnected.

**Comment by romgrm** [comment]
Hi @JlUgia :) 
I'll come back to this issue, after various investigations on the Flutter side, we found that replacing the PKPaymentButton() with a simple UIButton worked. 

So the issue comes from the PKPaymentButton, probably a communication problem between it and Flutter from iOS 16+.

Has the component changed recently? 

![issue_applepay](https://github.com/google-pay/flutter-plugin/assets/72022928/73e3da11-cdfc-410c-8ffe-3226081ad882)
